skimage/filters/ridges.py

```diff
@@ -106,8 +108,278 @@ def _check_sigmas(sigmas):
+                                    truncate=4.0):
+    """Multidimensional Gaussian filter with FFT-based Gaussian kernel.
+    Parameters
+    ----------
+    %(input)s
+    sigma : scalar or sequence of scalars
+        Standard deviation for Gaussian kernel. The standard
+        deviations of the Gaussian filter are given for each axis as a
+        sequence, or as a single number, in which case it is equal for
+        all axes.
+    order : int or sequence of ints, optional
+        The order of the filter along each axis is given as a sequence
+        of integers, or as a single number. An order of 0 corresponds
+        to convolution with a Gaussian kernel. A positive order
+        corresponds to convolution with that derivative of a Gaussian.
+    %(output)s
+    %(mode_multiple)s
+    %(cval)s
+    truncate : float
+        Truncate the filter at this many standard deviations.
+        Default is 4.0.
+    Returns
+    -------
+    gaussian_filter : ndarray
+        Returned array of same shape as `input`.
+    Notes
+    -----
+    The multidimensional filter is implemented as a sequence of
+    1-D convolution filters. The intermediate arrays are
+    stored in the same data type as the output. Therefore, for output
+    types with a limited precision, the results may be imprecise
+    because intermediate results may be stored with insufficient
+    precision.
+    Examples
+    --------
+    >>> from skimage.filters.ridges import gaussian_filter_with_FFT_kernel
+    >>> a = np.arange(50, step=2).reshape((5,5)).astype(float)
+    >>> a
+    array([[ 0.  2.  4.  6.  8.]
+           [10. 12. 14. 16. 18.]
+           [20. 22. 24. 26. 28.]
+           [30. 32. 34. 36. 38.]
+           [40. 42. 44. 46. 48.]])
+    >>> gaussian_filter_with_FFT_kernel(a, sigma=1)
+    array([[ 5.12436162,  6.40596063,  8.27030135, 10.13464207, 11.41624108],
+           [11.53235669, 12.8139557 , 14.67829642, 16.54263714, 17.82423615],
+           [20.85406027, 22.13565928, 24.        , 25.86434072, 27.14593973],
+           [30.17576385, 31.45736286, 33.32170358, 35.1860443 , 36.46764331],
+           [36.58375892, 37.86535793, 39.72969865, 41.59403937, 42.87563838]])
+    >>> from scipy import misc
+    >>> import matplotlib.pyplot as plt
+    >>> fig = plt.figure()
+    >>> plt.gray()  # show the filtered result in grayscale
+    >>> ax1 = fig.add_subplot(121)  # left side
+    >>> ax2 = fig.add_subplot(122)  # right side
+    >>> ascent = misc.ascent()
+    >>> result = gaussian_filter_with_FFT_kernel(ascent, sigma=5)
+    >>> ax1.imshow(ascent)
+    >>> ax2.imshow(result)
+    >>> plt.show()
+    """
+    input = np.asarray(input)
+    output = ndi._ni_support._get_output(output, input)
+    orders = ndi._ni_support._normalize_sequence(order, input.ndim)
+    sigmas = ndi._ni_support._normalize_sequence(sigma, input.ndim)
+    modes = ndi._ni_support._normalize_sequence(mode, input.ndim)
```

Comment:
This `_ni_support.py` helper module is not part of the public SciPy API and could in principle change without warning. Given that, we should not use it in scikit-image. We can keep a local copy with these helper functions if needed since the license is compatible.

Reply:
Okay, I will do that! Thanks for the suggestion, that makes a lot of sense.